Source/WebKit/Shared/WebProcessCreationParameters.cpp

```diff
@@ -526,6 +527,11 @@ bool WebProcessCreationParameters::decode(IPC::Decoder& decoder, WebProcessCreat
     if (!useDMABufSurfaceForCompositing)
         return false;
     parameters.useDMABufSurfaceForCompositing = WTFMove(*useDMABufSurfaceForCompositing);
+    std::optional<String> renderDeviceFile;
+    decoder >> renderDeviceFile;
+    if (!renderDeviceFile)
+        return false;
+    parameters.renderDeviceFile = WTFMove(*renderDeviceFile);
```

Comment:
It would nicer to pass a `dup()`'d file descriptor and then instantiate with `gdb_create_device(int fd)` in the WebProcess, which should be less finicky than using strings and would avoid needing access to `/dev/dri/*` in the WebProcess... it's a bit of a pity that the EGL extension to query the render node cannot return a file descriptor.

Oh, well, that's life 🤷🏼

Reply:
I think the web process needs access to /dev/dri in any case. Note that in case of swrast we don't have a device, I think it's clearer to pass an empty device file path than a null fd.

Reply:
File path is okay, this was commentary of things that came to mind, but nothing that IMO would block landing the patch at all 😉